src/Illuminate/Routing/AbstractRouteCollection.php

```diff
@@ -28,7 +28,7 @@ abstract class AbstractRouteCollection implements Countable, IteratorAggregate,
      */
     protected function handleMatchedRoute(Request $request, $route)
     {
-        if (! is_null($route)) {
+        if (! is_null($route) && $route->matches($request)) {
```

Comment:
This may be performing unnecessary "re-validations" for route matching.

I'm not sure if we should add something like a `$route->passes($request)` method for executing *only* the callback validator? Let me know if you have an opinion on this 👍

Reply:
It was necessary to execute the `CallbackValidator` to deny the route and throw the `NotFound` HTTP exception. Otherwise, the route matches and bypasses the validator.